pkg/stanza/fileconsumer/reader_factory.go

```diff
@@ -93,10 +94,12 @@ func (b *readerBuilder) build() (r *Reader, err error) {
 		Offset:       b.offset,
 	}
 
-	if b.splitter != nil {
-		r.splitter = b.splitter
+	var splitter *helper.Splitter
+	if b.splitFunc != nil {
+		r.splitFunc = b.splitFunc
 	} else {
-		r.splitter, err = b.splitterConfig.Build(false, b.readerConfig.maxLogSize)
+		splitter, err = b.splitterConfig.Build(false, b.readerConfig.maxLogSize)
+		r.splitFunc = splitter.SplitFunc
```

Comment:
Are we using the `Splitter` struct anywhere else? If not, perhaps we should delete it and change `splitterConfig.Build` to return the `SplitFunc` directly.

Reply:
I think there is no other place to use it. 
`splitterConfig.Build` does not directly return `SplitFunc`, because I want to replace `splitterConfig` with the interface `splitterFactory` in the next PR.

Reply:
I wonder why `encoding` and `flusher` were embedded in `Splitter` before, I thought their responsibilities were part of `Reader` but not part of `Splitter`. Thank you.

Reply:
All of these things got a little tangled up when they were added. I'm glad we are cleaning them up now. Thanks for your work on this.

Reply:
My pleasure.

Reply:
> I wonder why `encoding` and `flusher` were embedded in `Splitter` before, I thought their responsibilities were part of `Reader` but not part of `Splitter`. Thank you.

One thing I remembered about this is that multiline is used in `tcp` and `udp` operators as well, so I think encoding and flusher were combined into the same struct for convenience. We just need to make sure we do not break tcp/udp.

Reply:
>  I think encoding and flusher were combined into the same struct for convenience.

`tcp` and `udp` operators does not use `Splitter`, but directly uses `Encoding` and `SplitFunc`.
```Go
// Input is an operator that listens to a socket for log entries.
type Input struct {
	buffer []byte
	helper.InputOperator
	address       *net.UDPAddr
	addAttributes bool

	connection net.PacketConn
	cancel     context.CancelFunc
	wg         sync.WaitGroup

	encoding  helper.Encoding
	splitFunc bufio.SplitFunc
	resolver  *helper.IPResolver
}
```
I don't think this will bring breaking changes.

Reply:
👍  Thanks for looking into it.